File: adapter/src/main/java/com/android/jmaxime/annotations/BindLayoutRes.java
package com.android.jmaxime.annotations;

import android.support.annotation.LayoutRes;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target({ElementType.TYPE, ElementType.PARAMETER})
@Retention(RetentionPolicy.RUNTIME)
public @interface BindLayoutRes {
    /**
     * View ID to which the field will be bound.
     */
    @LayoutRes int value();
}

File: adapter/src/main/java/com/android/jmaxime/annotations/BindLayoutRes.kt
package com.android.jmaxime.annotations

import android.support.annotation.LayoutRes

/**
 * @author Maxime Jallu
 * @since 09/06/2017
 * Use this Class for : <br></br>
 * ... {DOCUMENTATION}
 */
//@Scope
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE, AnnotationTarget.VALUE_PARAMETER)
@Retention(AnnotationRetention.RUNTIME)
annotation class BindLayoutRes(
        /**
         * View ID to which the field will be bound.
         */
        @LayoutRes val value: Int
)
